### DISCUSSION TITLE
[Help] Inject gollum-auth in Docker

### DISCUSSION THREAD

**Discussion by nwithan8** [gollum/gollum#2087] [Q&A] [answered]
Hello, I'm trying to add [gollum-auth](https://github.com/bjoernalbers/gollum-auth) via the `config.ru` (confused between this and the `config.rb` file) for Rake. Problem is, I'm running Gollum from a Docker container, so any installation of gemfiles will be overwritten on each reboot, and the Docker container boots with Gollum already running inside it, so I can't see any easy way to add the necessary gems and restart Gollum.

**Comment by dometto** [comment]
I took the liberty of transferring this to a discussion.

The fundamental issue here is that gollum-auth, and also [omnigollum](https://github.com/arr2036/omnigollum/), are distinct projects from gollum. So we cannot really publish ready-made Docker images that contain them, since we don't have control over the code. This raises the question how we can best offer support for authentication in Gollum; I'll open a separate discussion for that. I'll post a solution outline for your problem in the next comment,

**Reply by dometto** [reply]
See https://github.com/gollum/gollum/discussions/2088

**Comment by dometto** [comment, marked as answer]
The best solution to your problem may be to create a custom Dockerfile based on the official gollum image. Something like:

```bash
$ cat Dockerfile 
FROM gollumwiki/gollum:6
USER root
RUN gem install gollum-auth
USER www-data
```

Then e.g. `docker build -t gollum-auth .`
